SABnzbd.py

```diff
@@ -1539,6 +1539,23 @@ def main():
 
         # 30 sec polling tasks
         if not timer % 10:
+            servers = config.get_servers()
+            for srv in config.get_servers():
+                server = servers[srv]
+                if server.expire_date():
+                    if time.time() > time.mktime(time.strptime(server.expire_date(), "%Y-%m-%d")):
+                        logging.warning("Server %s is expiring", server.displayname())
+                        server.expire_date.set("")
+                        config.save_config()
+                if server.quota_left():
+                    if (
+                        server.quota_left.get_float() + server.usage_at_start.get_float()
+                        < sabnzbd.BPSMeter.grand_total[srv]
+                    ):
+                        logging.warning("Server %s has used the spcified quota", server.displayname())
```

Comment:
I would have expected the code to be at the same spot as the regular quota check in the BPSMeter?

Reply:
I wanted the performance impact to be minimal because all that happens is a warning in the program. Unless the user is monitoring it at the moment it happens, sending the warning at exactly the right moment or 5 minutes later makes no difference.